Added clean build folder (#2280)
swiftlang/vscode-swift: userdocs/userdocs.docc/Articles/Features/automatic-task-creation.md

- **Build All**: Build all targets in the Package.
- **Build Debug \<Executable\>**: Each executable product in a Package.swift get a task for building a debug build.
- **Build Release \<Executable\>**: Each executable product in a Package.swift get a task for building a release build.

> - **Clean Build**: Runs `swift package clean` to remove all build artifacts.

Reviewer: Can we rename this to match the command name? You'll also have to change `SwiftTaskProvider.cleanBuildName` and the examples below to match:
```suggestion
- **Clean Build Folder**: Runs `swift package clean` to remove all build artifacts.
```

- **Build Debug \<Library\>**: Each library product in a Package.swift get a task for building a debug build.
- **Build Release \<Library\>**: Each library product in a Package.swift get a task for building a release build.

> ⚠️ Important: Tasks will not be created for automatic library products, as you cannot specify a `--product` option for automatic products when building. For more information see the [Swift Package Manager documentation for Product definitions](https://docs.swift.org/package-manager/PackageDescription/PackageDescription.html#product).

## Combining tasks

> Automatic tasks can be composed into a single workflow using the `dependsOn` property. The following `tasks.json` runs a clean before building:

Reviewer: ```suggestion
You can compose multiple built-in Swift tasks into a single [compound task](https://code.visualstudio.com/docs/debugtest/tasks#_compound-tasks) in your `.vscode/tasks.json`. For example, you can create a `Clean Rebuild` compound task that runs the `swift: Clean Build Folder` task followed by the `swift: Build All` task:
```

```json
{

>     "version": "2.0.0",
>     "tasks": [
>         {
>             "label": "Clean Rebuild",
>             "dependsOn": ["Clean Build Folder", "Build All"],
>             "dependsOrder": "sequence",
>             "group": {"kind": "build", "isDefault": true},
>             "problemMatcher": []
>         }
>     ]
> }

Reviewer: The task labels generated by VS Code include a `swift:` prefix:

```suggestion
{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "Clean Rebuild",
            "dependsOn": ["swift: Clean Build Folder", "swift: Build All"],
            "dependsOrder": "sequence",
            "group": "build"
        }
    ]
}
```

```

> The same approach works for debug configurations via `preLaunchTask` in `launch.json`:

Reviewer: ```suggestion
[VS Code Debug Configurations](https://code.visualstudio.com/docs/debugtest/debugging-configuration) have a `preLaunchTask` property that can be used to run a single task before launching the debugger. You can use a compound task to run multiple smaller tasks as a `preLaunchTask`. For example, if you add the following to your `.vscode/launch.json`, VS Code will run the `Clean Rebuild` compound task before launching the debugger:
```

```json

> {
>     "configurations": [
>         {
>             "type": "swift",
>             "request": "launch",
>             "name": "Launch MyApp",
>             "target": "MyApp",
>             "preLaunchTask": "Clean Build Folder"
>         }
>     ]
> }

Reviewer: We should have this use the `Clean Rebuild` task from the example above:

```suggestion
{
    "configurations": [
        {
            "type": "swift",
            "request": "launch",
            "name": "Launch MyApp",
            "target": "MyApp",
            "preLaunchTask": "Clean Rebuild"
        }
    ]
}
```

```

> For more information, see the [VS Code Tasks](https://code.visualstudio.com/docs/editor/tasks) and [preLaunchTask](https://code.visualstudio.com/docs/debugtest/debugging-configuration) documentation.

Reviewer: Was looking at the rest of the documentation and noticed that we already link to both of these pages in various notes. I think we can drop this sentence entirely.

> For more information, see the [VS Code Tasks](https://code.visualstudio.com/docs/editor/tasks) and [preLaunchTask](https://code.visualstudio.com/docs/editor/debugging#_launchjson-attributes) documentation.

Reviewer: The debugging page you linked doesn't have a `_launchjson-attributes` anchor. It also doesn't mention anything about `preLaunchTask`. Maybe we should just link to https://code.visualstudio.com/docs/debugtest/debugging-configuration?

Author: Yes, I will change Link to debugging-configuration.
